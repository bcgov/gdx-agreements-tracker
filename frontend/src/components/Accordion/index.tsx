import * as React from "react";
import { Accordion as MUIAccordion } from "@mui/material/";
import AccordionSummary from "@mui/material/AccordionSummary";
import AccordionDetails from "@mui/material/AccordionDetails";
import Typography from "@mui/material/Typography";
import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
import { styled } from "@mui/material";
import bcgovTheme from "../../bcgovTheme";

export const Accordion = ({
  children,
  sectionTitle,
}: {
  children: JSX.Element | JSX.Element[];
  sectionTitle: string;
}) => {
  const StyledAccordionSummary = styled(AccordionSummary)({
    backgroundColor: "#f3f3f3",
  });

  const StyledTypography = styled(Typography)({
    color: bcgovTheme.palette.primary.main,
    fontWeight: "bold",
  });

  return (
    <MUIAccordion defaultExpanded>
      <StyledAccordionSummary
        expandIcon={<ExpandMoreIcon />}
        aria-controls="panel-content"
        aria-label="button"
      >
        <StyledTypography>{sectionTitle}</StyledTypography>
      </StyledAccordionSummary>
      <AccordionDetails>{children}</AccordionDetails>
    </MUIAccordion>
  );
};
