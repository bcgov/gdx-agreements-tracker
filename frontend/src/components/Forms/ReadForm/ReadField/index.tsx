import {
  Checkbox,
  Chip,
  Divider,
  FilledInput,
  FormControl,
  InputLabel,
  List,
  ListItem,
  ListItemButton,
  ListItemIcon,
  ListItemText,
  ListSubheader,
  Select,
  Stack,
  TextField,
  Typography,
} from "@mui/material";
import { IReturnValue } from "types";
import { GridItem } from "../../FormLayout/GridItem";
import { useNavigate } from "react-router-dom";
import DescriptionIcon from "@mui/icons-material/Description";
/**
 * Represents a read-only field component.
 *
 * @param   {object}       props        - The properties object.
 * @param   {string}       props.width  - The width of the field.
 * @param   {string}       props.title  - The title of the field.
 * @param   {IReturnValue} props.value  - The value to display.
 * @param   {string}       [props.type] - The type of the field.
 * @returns {JSX.Element}               The rendered ReadField component.
 */

export const ReadField = ({
  width,
  title,
  value,
  type,
}: {
  width: string;
  title: string;
  value: IReturnValue;
  type?: string;
}) => {
  const navigate = useNavigate();
  const renderer = () => {
    switch (type) {
      case "multiSelect":
        return (
          <FormControl fullWidth>
            <InputLabel>{title}</InputLabel>
            <Select
              disabled
              multiple
              value={value}
              IconComponent={() => null} //To hide dropdown arrow
              input={
                <FilledInput
                  sx={{
                    "& .MuiInputBase-input.Mui-disabled": {
                      WebkitTextFillColor: "#333",
                      background: "#fff",
                    },
                    "& .MuiInputBase-root": {
                      WebkitTextFillColor: "#333",
                      background: "#fff!important",
                    },
                  }}
                />
              }
              renderValue={(selected) => (
                <Stack gap={1} direction="row" flexWrap="wrap">
                  {(value as string[]).map((value) => (
                    <Chip key={value} label={value} />
                  ))}
                </Stack>
              )}
            ></Select>
          </FormControl>
        );
      case "checkbox":
        return (
          <div>
            <Typography variant="subtitle1" color="textSecondary">
              {title}:
            </Typography>
            <Checkbox disabled checked={value as boolean} />
          </div>
        );

      case "link":
        return (
          <List
            sx={{ width: "100%", maxWidth: "13rem", bgcolor: "#fbfbfb" }}
            subheader={
              <ListSubheader sx={{ bgcolor: "#2d2b2b", color: "#fff" }}>{title}</ListSubheader>
            }
          >
            {(value as unknown as { link: string; label: string }[]).map((value) => {
              return (
                <>
                  <Divider />
                  <ListItem component="div" disablePadding>
                    <ListItemButton onClick={() => navigate(value.link)}>
                      <ListItemText sx={{ color: "blue" }} primary={`${value.label}`} />
                      <ListItemIcon sx={{ color: "inherit" }}>
                        <DescriptionIcon />
                      </ListItemIcon>
                    </ListItemButton>
                  </ListItem>
                </>
              );
            })}
          </List>
        );

      default:
        return (
          <TextField
            disabled
            label={title}
            value={value}
            fullWidth
            sx={{
              "& .MuiInputLabel-root ": {
                WebkitTextFillColor: "#16008fc4",
              },
              "& .MuiInputBase-input.Mui-disabled": {
                WebkitTextFillColor: "#333",
                background: "#fff",
              },
              "& .MuiInputBase-root": {
                WebkitTextFillColor: "#333",
                background: "#fff!important",
              },
            }}
            id="filled-size-small"
            variant="filled"
            size="small"
            multiline
          />
        );
    }
  };

  return <GridItem width={width}>{renderer()}</GridItem>;
};
