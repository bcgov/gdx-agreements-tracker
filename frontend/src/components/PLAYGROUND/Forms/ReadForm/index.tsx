import React from "react";

export const ReadForm = () => {
  return <div>ReadForm</div>;
};
