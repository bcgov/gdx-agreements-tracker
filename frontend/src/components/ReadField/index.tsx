import styled from "@emotion/styled";
import { Paper } from "@mui/material";
import React from "react";
import { GridItem } from "../GDXForm/FormLayout/GridItem";

export const ReadField = ({
  width,
  title,
  value,
}: {
  width: string;
  title: string;
  value: string | number;
}) => {
  const StyledPaper = styled(Paper)({
    height: "auto",
    flex: 1,
    display: "flex",
    overflow: "auto",
  });

  return (
    <GridItem width={width}>
      <StyledPaper>
        <h3>{title}</h3>
        <p>{value}</p>
      </StyledPaper>
    </GridItem>
  );
};
