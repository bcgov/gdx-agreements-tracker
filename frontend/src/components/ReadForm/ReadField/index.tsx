import styled from "@emotion/styled";
import { Checkbox, Paper } from "@mui/material";
import React from "react";
import { IReturnValue } from "types";
import { GridItem } from "../../GDXForm/FormLayout/GridItem";

export const ReadField = ({
  width,
  title,
  value,
}: {
  width: string;
  title: string;
  value: IReturnValue;
}) => {
  const renderer = () => {
    switch (typeof value) {
      case "boolean":
        return (
          <>
            <h3>{title}</h3>
            <Checkbox disabled checked={value} />
          </>
        );
      default:
        return (
          <>
            <h3>{title}</h3>
            <p>{Array.isArray(value) ? value.join(" | ") : value}</p>
          </>
        );
    }
  };

  const StyledPaper = styled(Paper)({
    height: "auto",
    flex: 1,
    display: "flex",
    overflow: "auto",
  });

  return (
    <GridItem width={width}>
      <StyledPaper>{renderer()}</StyledPaper>
    </GridItem>
  );
};
