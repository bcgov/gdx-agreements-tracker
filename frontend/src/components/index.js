export { ButtonNav } from "./ButtonNav";
export { Debug } from "./Debug";
export { Loader } from "./Loader";
export { Main } from "./Layout/Main";
export { PageLink } from "./PageLink";
export { RowEditDeleteChip } from "./RowEditDeleteChip";
export { Sidebar } from "./Sidebar";
export { DetailsList } from "./DetailsList";
export { Table } from "./Table";
