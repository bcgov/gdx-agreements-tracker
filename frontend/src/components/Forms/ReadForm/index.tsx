import React from "react";
import { IReturnValue } from "types";
import { ReadField } from "./ReadField";
import { FormLayout } from "components/Forms/FormLayout";

export const ReadForm = ({
  fields,
}: {
  fields: { width: string; title: string; value: IReturnValue }[];
}) => {
  return (
    <FormLayout>
      {fields.map(({ width, title, value }) => {
        console.log('value', value)
        return <ReadField width={width} title={title} value={value} key={title}></ReadField>;
      })}
    </FormLayout>
  );
};
