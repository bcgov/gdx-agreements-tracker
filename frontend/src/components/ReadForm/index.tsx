import React from "react";
import { FormLayout } from "../GDXForm";
import { ReadField } from "./ReadField";

export const ReadForm = ({
  fields,
}: {
  fields: { width: string; title: string; value: string | number }[];
}) => {
  return (
    <FormLayout>
      {fields.map(({ width, title, value }) => {
        return <ReadField width={width} title={title} value={value}></ReadField>;
      })}
    </FormLayout>
  );
};
