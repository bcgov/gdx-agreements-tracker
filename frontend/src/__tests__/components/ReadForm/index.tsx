import { QueryClient, QueryClientProvider } from "react-query";
import { BrowserRouter } from "react-router-dom";
import { ReadForm } from "../../../components/ReadForm";
import TestRenderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";
import { mount, shallow } from "enzyme";
import { ReadField } from "components/ReadForm/ReadField";

const fields = [
  {
    width: "full",
    title: "Mock Title",
    value: "Mock Value",
  },
  {
    width: "half",
    title: "Mock Title 2",
    value: 2,
  },
];

describe("Renders <ReadForm /> component", () => {
  const wrapper = shallow(<ReadForm fields={fields} />);

  it("renders the ReadForm component", () => {
    expect(wrapper.html()).toMatchSnapshot();
    // expect((wrapper.props().children[0].props.title)).toEqual("Mock Title");
  });
});
