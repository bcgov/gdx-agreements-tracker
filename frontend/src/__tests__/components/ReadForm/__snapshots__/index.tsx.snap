// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <ReadForm /> component renders the ReadForm component 1`] = `"<div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12 css-1idn90j-MuiGrid-root\\"><div class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-fitgaq-MuiPaper-root\\"><h3>Mock Title</h3><p>Mock Value</p></div></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 css-1fyyp8j-MuiGrid-root\\"><div class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-fitgaq-MuiPaper-root\\"><h3>Mock Title 2</h3><p>2</p></div></div></div>"`;

exports[`Renders <ReadForm /> component renders the ReadForm component 2`] = `"<div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12 css-1idn90j-MuiGrid-root\\"><div class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-fitgaq-MuiPaper-root\\"><h3>Mock Title</h3><p>Mock Value</p></div></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 css-1fyyp8j-MuiGrid-root\\"><div class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-fitgaq-MuiPaper-root\\"><h3>Mock Title 2</h3><p>2</p></div></div></div>"`;
